### PR TITLE
Courses: update UI for course catalog & hoc options

### PR DESCRIPTION
**Pull request by breville**
#### signed-in teacher, english
_teacher english

#### signed-in student, english
courses student en

#### signed out, english
_signedout english

#### signed-in teacher, non-english
_teacher spanish

#### signed-in student, non-english
courses student nonen

#### signed out, non-english
_signedout spanish

### CODE REVIEW

**Comment by breville** [comment]
cc @tanyaparker

**Comment by tanyaparker** [comment]
Erin is out so can you assign another dev to review?

Is the Local Classes box localized? Ideally I want this to only show for US users (whether they are in English or non-EN). It's ok if it shows up for non-EN now as long as it's US only AND it's localized. Otherwise, if we can't target by country OR if it's not localized, I would rather hide it completely for all non-EN.

Also, is the Hour of Code widget localized? I don't want to switch it out just yet, but in some time (after we get some translations) we can replace the 8 block version with the 4 block version so we have fewer versions/more consistency.

File: shared/haml/course_explorer_table.haml [hunk 1]
```diff
@@ -56,7 +56,7 @@
             - unless course[:starts] == 0
               %div{class: "bar-#{course[:starts]}"}
                 &nbsp;
-            %div{class: "courserow bar-#{course[:ends] - course[:starts] + 1}", style: "background-color: #{"#1AC7D6" || course[:color] || '#00adbc'}"}
+            %div{class: "courserow bar-#{course[:ends] - course[:starts] + 1}", style: "background-color: #{"#00adbc"}"}
```
Comment (tanyaparker) [inline shared/haml/course_explorer_table.haml]: I like sticking to our color palette a lot better. 👍

File: apps/i18n/common/en_us.json [hunk 2]
```diff
@@ -492,13 +492,15 @@
   "teacherCommunity": "Teacher Community",
   "teacherCommunityDescription": "Ask questions about curriculum, share ideas from your lessons, and get help from other teachers",
   "teacherCourseHoc": "Hour of Code",
-  "teacherCourseHocDescription": "Celebrated in December, but available year-round, the Hour of Code makes computer science fun and accessible to all ages.",
+  "teacherCourseHocDescription": "If you don’t have time for a full length course, try a one-hour tutorial designed for all ages. Join millions of students and teachers in over 180 countries by starting with an Hour of Code.",
```
Comment (tanyaparker) [inline apps/i18n/common/en_us.json:502]: Can we link [Hour of Code](https://hourofcode.com)?

File: apps/src/templates/studioHomepages/TeacherAssignablesCatalog.jsx [hunk 3]
```diff
@@ -1,52 +1,78 @@
...
+        </ContentContainer>
+
+        <ContentContainer
+          heading={i18n.teacherCourseHoc()}
+          description={i18n.teacherCourseHocDescription()}
           isRtl={isRtl}
-        />
-      </ContentContainer>
+          linkText={i18n.teacherCourseHocLinkText()}
+          link={`${codeOrgUrlPrefix}/hourofcode/overview`}
```
Comment (tanyaparker) [inline apps/src/templates/studioHomepages/TeacherAssignablesCatalog.jsx:71]: I'm not sure I'm reading this right but wanted to confirm teachers go to code.org/hourofcode/overview but students go to code.org/learn?

File: apps/i18n/common/en_us.json (hunk 2 above)
Comment (breville) [reply apps/i18n/common/en_us.json:502]: @Bjvanminnen I tried once and failed.. do we have a good way to embed links in apps strings, wrapping the link around some text that's in the string?

File: apps/src/templates/studioHomepages/TeacherAssignablesCatalog.jsx (hunk 3 above)
Comment (breville) [reply apps/src/templates/studioHomepages/TeacherAssignablesCatalog.jsx:71]: Signed-in student in English gets /learn.
Signed-in teacher in English gets /hourofcode/overview.

File: apps/i18n/common/en_us.json (hunk 2 above)
Comment (Bjvanminnen) [reply apps/i18n/common/en_us.json:502]: We don't have a good way to do it inside of translated strings. The approach I usually aim for is to have a link following the string.

File: apps/src/templates/studioHomepages/Courses.jsx [hunk 4]
```diff
@@ -165,8 +161,24 @@ const Courses = React.createClass({
...
+            showLink={true}
+          >
+            <ProtectedStatefulDiv ref="hocCoursesFirstRow"/>
+          </ContentContainer>
+        )}
+
+        {/* Anything but a teacher or student in English.
+            That is: signed-out, or a student or teacher in non-English. */}
+        {(isSignedOut || !isEnglish) && (
+          <div id="all-courses">
```
Comment (Bjvanminnen) [inline apps/src/templates/studioHomepages/Courses.jsx]: Does this div need an id? I don't see it used anywhere (we do look for a class by this name in componentDidMount)

File: shared/haml/course_explorer_table.haml (hunk 1 above)
Comment (Bjvanminnen) [reply shared/haml/course_explorer_table.haml]: presumably we don't need the `#{}` portion now that there's no ruby in there (just a string)

File: apps/i18n/common/en_us.json (hunk 2 above)
Comment (breville) [reply apps/i18n/common/en_us.json:502]: @tanyaparker Can we rewrite the content to make that work?

File: apps/src/templates/studioHomepages/Courses.jsx (hunk 4 above)
Comment (breville) [reply apps/src/templates/studioHomepages/Courses.jsx]: Oh, I just added this to make my own visual inspection of the DOM a bit easier.  I can take it back out again if not a good practice.

**Comment by breville** [comment]
@tanyaparker:

Local Classes box is localised.  I'm waiting to hear back from infrastructure folks about perf impact of calling Geocoder on every page, so in the meantime I'll just make it EN only for now.

The Hour of Code content is also localised.  The boxes themselves are the legacy HAML.

File: apps/src/templates/studioHomepages/Courses.jsx (hunk 4 above)
Comment (Bjvanminnen) [reply apps/src/templates/studioHomepages/Courses.jsx]: When I'm reading JSX and I see an id, my first thought is "that means there are side effects somewhere else that I need to track down" (either CSS, or jquery doing something). To some extent it's conceptually similar to adding a global variable (although arguably not quite egregious). As such, I prefer avoiding them unless absolutely necessary.

File: apps/i18n/common/en_us.json (hunk 2 above)
Comment (tanyaparker) [reply apps/i18n/common/en_us.json:502]: Maybe append "Visit HourOfCode.com to learn more." at the end where the whole thing is hyperlinked? Is that what you mean?

**Pull request merged**